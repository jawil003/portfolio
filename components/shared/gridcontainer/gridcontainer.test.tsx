import { render } from "@testing-library/react";
import {
  GridContainer,
  GridContainerProps,
} from ".";

const variants: GridContainerProps[] = [
  {
    alignContent: "center",
    alignItems: "center",
    justifyContent: "center",
    justifyItems: "center",
  },
  {
    columnGap: "10px",
    rowGap: "10px",
  },
];

describe("GridContainer", () => {
  test.each(variants)(
    "Create GridContainer with %p",
    (variant) => {
      const testgridcontainer = render(
        <GridContainer {...variant} />,
      );

      expect(
        testgridcontainer.container
          .firstChild,
      ).toMatchSnapshot();
    },
  );
});
