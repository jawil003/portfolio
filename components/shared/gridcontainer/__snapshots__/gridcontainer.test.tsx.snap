// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridContainer Create GridContainer with {"alignContent": "center", "alignItems": "center", "justifyContent": "center", "justifyItems": "center"} 1`] = `
.emotion-0 {
  display: grid;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0>* {
  width: 100%;
  height: 100%;
}

<body>
  <div>
    <div
      class="emotion-0"
    />
  </div>
</body>
`;

exports[`GridContainer Create GridContainer with {"columnGap": "10px", "rowGap": "10px"} 1`] = `
.emotion-0 {
  display: grid;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  row-gap: 10px;
}

.emotion-0>* {
  width: 100%;
  height: 100%;
}

<body>
  <div>
    <div
      class="emotion-0"
    />
  </div>
</body>
`;
